### ISSUE TITLE
Auth error prompt when cloning repo from nonexistent URL

### ISSUE THREAD

**Issue by kanishk98** [desktop/desktop#5661]
## Description

On trying to clone a repo using an incorrect URL, Desktop raises an authentication error instead of indicating that the repo wasn't found in the user's profile. 

## Version

* GitHub Desktop: 1.3.5

* Operating system: macOS High Sierra 10.13.6

## Steps to Reproduce

1. Click on Clone repository. 
2. Navigate to the URL section. 
3. Enter an invalid URL. 

### Expected Behavior

Desktop raises an error prompt stating that the provided repo URL is invalid. 

### Actual Behavior

Desktop goes into a loop of error prompts related to authentication. If correct username and password is provided, then the same prompt is raised again.

## Additional Information

![screen shot 2018-09-16 at 5 07 09 pm](https://user-images.githubusercontent.com/24617297/45596367-9e04f680-b9d8-11e8-9972-22e032fd214d.png)
![screen shot 2018-09-16 at 5 07 27 pm](https://user-images.githubusercontent.com/24617297/45596369-a0ffe700-b9d8-11e8-9302-22da0fc1094b.png)

**Comment by kanishk98** [comment]
If the Desktop core team can give me some basic info on this issue, I'd love to work on the same.

**Comment by shiftkey** [comment]
@kanishk98 let me look into reproducing the issue on my end and figure out what changes need to be made

**Comment by shiftkey** [comment]
Actually, I need to drop myself from this because I don't have the bandwidth. Apologies!

**Comment by steveward** [comment]
I was able to reproduce this issue, but after looking through the codebase I'm going to have to defer to someone on @desktop/eng to clarify what exactly needs to be changed. 😄 

Here's what I tested:

- Full GitHub URL pointing to nonexistent repo (i.e. https://github.com/steveward/norepohere)
Result: Gets stuck in the `Authentication Failed` loop.

- Shortened `user/repo` option that points to a nonexistent repository.
Result: Correctly throws the `fatal: repository does not exist` error.

- GitHub SSH URL (i.e. `git@github.com:steveward/norepohere`)
Result: Correctly throws the `authentication failed` error once due to the repo not existing.

- Non-GitHub URL pointing to a repository that does not exist (i.e. any web site).
Result: Correctly errors out.

**Comment by kanishk98** [comment]
@shiftkey no problem, thanks for trying to help out. 
@steveward  thanks a ton for your research. I was looking at [lib/git/clone.ts](https://github.com/desktop/desktop/blob/master/app/src/lib/git/clone.ts) just now and I'll start by checking out the objects returned from the promises in different cases. Maybe that'll help the Desktop team too. 

Apologies for the slow response time, got exams going on. :)

**Comment by billygriffin** [comment]
No apology necessary @kanishk98, we really appreciate your willingness to dig into the code and hopefully make some forward progress! ❤️

**Comment by kanishk98** [comment]
Okay, I've found a few interesting results. The `app/src/lib/logging/renderer/install.ts` file logs that the account is found if the credentials are valid (line 31, refer to the screenshot below) and line 23 of the same file says repo not found. This error is then sent forward to the main process and `generic-git-auth.tsx` raises an error dialog before the `clone-repository.tsx` file can check the value of `url` variable and raise an appropriate error message. 

Screenshot of console:

![image](https://user-images.githubusercontent.com/24617297/46026065-0bff9b00-c108-11e8-8aa5-8d95a7215ce4.png)

This code in `clone-repository.tsx` doesn't get executed: 
```
if (!url) {
      console.log('raising error');
      const error = new Error(
        `We couldn't find that repository. Check that you are logged in, the network is accessible, and the URL or repository alias are spelled correctly.`
      )
      this.setState({ loading: false, error })
      return
    }

```

I'm not sure of how `ipcRenderer` works here and how I can fix this issue just yet, but I'll keep looking. Just figured this might be of interest. 

@billygriffin thanks for the support!

**Comment by kanishk98** [comment]
Okay, I messed around with the code here and got the desired functionality to work. 

![image](https://user-images.githubusercontent.com/24617297/46489378-d56e0280-c822-11e8-90e2-2b1024bb27e0.png)

The changed code is in `src\ui\clone-repository\clone-repository.tsx`, and I simply added a `fetch()` method to check for HTTP error codes.

```
if (identifier && account) {
      const repoExists = await fetch(url)
      if (repoExists.status == 404) {
        return null
}
```

It'd be great if other contributors could let me know if there are any other problems with this approach. I'll submit a PR after waiting for a bit.
